br/tests/lightning_fail_fast_on_nonretry_err/run.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/sh
+
+set -eu
+
+run_sql 'DROP DATABASE IF EXISTS csv'
+
+out_file_name="$TEST_DIR/sql_res.$TEST_NAME.txt"
+export GO_FAILPOINTS="github.com/pingcap/tidb/br/pkg/lightning/backend/local/WriteToTiKVNotEnoughDiskSpace=return(true)"
+start_time=$(date +'%s')
+run_lightning --backend local > $out_file_name 2>&1 || true
+export GO_FAILPOINTS=""
+check_contains "The available disk of TiKV"
+used_time=$(($(date +'%s') - $start_time))
+echo "time used to run lightning ${used_time}s"
+# writeAndIngestByRanges will retry for at least 810s on retryable error
+if [ $used_time -ge 60 ]; then
```

Comment:
Running a lightning with 810s is too long.  Consider use the `timeout` command.  In this way we even don't need to collect the start time and end time
```bash
timeout 30 run_lightning --backend local > "$out_file_name" 2>&1
lightning_exit_code=$?
case "$lightning_exit_code" in
    0)
        echo "lightning exits normally, but we expect it exit with an abnormal exit code" >&2
        exit 1
        ;;
    124)
        echo "lightning running too long, seems like it doesn't fail fast" >&2
        exit 1
        ;;
esac
```

Reply:
should not happen in this case since the error is not retryable now.

Reply:
Right, currently the timeout won't happen.  I just consider in the future, if someone changes the Lightning behavior unexpectedly, making the Lightning not fail fast for this test, the timeout mechanism can prevent the Lightning from running too long.

Reply:
mac don't has `timeout`(may need to install using brew)

will not change it, the case will fail in CI anyway, and the whole CI takes more than 30m, so we can wait.

Reply:
OK.  Got it